### PR TITLE
Identity manager/register flow

### PR DESCRIPTION
**Pull request by erick20**

### CODE REVIEW

File: src/IdentityManager/Repository.EntityFramework/Migrations/20210308183812_init.cs [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ﻿using System;
 using Microsoft.EntityFrameworkCore.Migrations;
 
-namespace Travely.IdentityManager.Repository.Migrations
+namespace Travely.IdentityManager.Repository.EntityFramework.Migrations
 {
-    public partial class CreateIdentityServerDB : Migration
+    public partial class init : Migration
```
Comment (GRustamyan-ST) [inline src/IdentityManager/Repository.EntityFramework/Migrations/20210308183812_init.cs:6]: Recreate migration and made it capital `init` -> `Init`, so we match C# code style in generated classes.

File: src/IdentityManager/WebApi/Controllers/AccountController.cs [hunk 2]
```diff
@@ -8,16 +8,21 @@
...
+using System.Collections.Generic;
+using Microsoft.AspNetCore.JsonPatch;
+using IdentityManager.WebApi.Extensions;
 
 namespace Travely.IdentityManager.WebApi.Controllers
 {
     [Route("api/[controller]")]
-    [ApiController]
+    [Produces("application/json")]
+    //[ApiController]
```
Comment (GRustamyan-ST) [inline src/IdentityManager/WebApi/Controllers/AccountController.cs:21]: Uncomment `ApiController`. 
https://www.strathweb.com/2018/02/exploring-the-apicontrollerattribute-and-its-features-for-asp-net-core-mvc-2-1/

File: src/IdentityManager/WebApi/Extensions/AutoMapperConfigurationExtension.cs [hunk 3]
```diff
@@ -1,19 +1,29 @@
...
     public static class AutoMapperConfigurationExtension
     {
         public static void ConfigureAutoMapper(this IServiceCollection services)
         {
-            
             services.AddAutoMapper(typeof(PersistGrantProfile));
+            services.AddAutoMapper(typeof(UserProfile));
+            services.AddAutoMapperBuilder(builder =>
+            {
+                builder.Profiles.Add(new UserProfile(services.BuildServiceProvider().GetRequiredService<IPasswordHasher<User>>()));
```
Comment (GRustamyan-ST) [inline src/IdentityManager/WebApi/Extensions/AutoMapperConfigurationExtension.cs:23]: It is bad building ServiceProvider manually. Better approach https://stackoverflow.com/a/44901316/2572312

Correct approach https://stackoverflow.com/a/49627570/2572312

File: src/IdentityManager/WebApi/Extensions/ServiceConfigurationExtension.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+namespace IdentityManager.WebApi.Extensions
+{
+    public static class ServiceConfigurationExtension
+    {
+        public static void ConfigureFilterServices(this IServiceCollection services)
+        {
+            services.AddControllers(config =>
+            {
+                config.Filters.Add(new ValidateModelStateActionFilter());
```
Comment (GRustamyan-ST) [inline src/IdentityManager/WebApi/Extensions/ServiceConfigurationExtension.cs:20]: I beleive ApiController attribute already do this for you

File: src/IdentityManager/WebApi/Mappers/UserProfile.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+                .ForMember(dest => dest.Name, opt => opt.MapFrom(src => src.AgencyName))
+                .AfterMap((src, dest) => dest.CreatedDate = DateTime.UtcNow)
+                .AfterMap((src, dest) => dest.LogoFile = "aa")
+                .AfterMap((src, dest) => dest.Address = "ad")
+                ;
+            
+            CreateMap<RegisterRequestModel, Employee>()
+                .ForMember(dest => dest.Email, opt => opt.MapFrom(src => src.Email))
+                .AfterMap((src, dest) => dest.CreatedDate = DateTime.UtcNow)
+                .AfterMap((src, dest) => dest.FirstName = "a") // remove after fix db nullable
```
Comment (GRustamyan-ST) [inline src/IdentityManager/WebApi/Mappers/UserProfile.cs:36]: Don't forget to remove.

File: src/IdentityManager/WebApi/Extensions/ServiceConfigurationExtension.cs (hunk 4 above)
Comment (erick20) [reply src/IdentityManager/WebApi/Extensions/ServiceConfigurationExtension.cs:21]: I add this for swagger, to fix the error responseModel in UI

**Pull request merged**